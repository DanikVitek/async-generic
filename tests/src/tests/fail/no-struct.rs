use async_generic::async_generic;

#[async_generic]
struct Struct {}

fn main() {}
