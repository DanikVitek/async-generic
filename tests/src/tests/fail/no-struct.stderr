error: async_generic can only be used with functions, traits or impls
 --> src/tests/fail/no-struct.rs:3:1
  |
3 | #[async_generic]
  | ^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `async_generic` (in Nightly builds, run with -Z macro-backtrace for more info)

error: expected `fn`
 --> src/tests/fail/no-struct.rs:4:1
  |
4 | struct Struct {}
  | ^^^^^^

error: expected `trait`
 --> src/tests/fail/no-struct.rs:4:1
  |
4 | struct Struct {}
  | ^^^^^^

error: expected `impl`
 --> src/tests/fail/no-struct.rs:4:1
  |
4 | struct Struct {}
  | ^^^^^^
